src/notification/NotificationController.ts

```diff
@@ -0,0 +1,266 @@
+import type { Patch } from 'immer';
+import { nanoid } from 'nanoid';
+
+import { hasProperty } from '../util';
+import { BaseController } from '../BaseControllerV2';
+
+import type { RestrictedControllerMessenger } from '../ControllerMessenger';
+
+/**
+ * @type NotificationState
+ * @property notifications - Stores existing notifications to be shown in the UI
+ */
+export type NotificationState = {
```

Comment:
What do you think about calling this NotificationControllerState instead? I have always disliked our arbitrary naming of state and other accompanying types in controllers. It seems like it'd make it really difficult for consumers who want to import this type to remember the exact name of the type. If we consistent across controllers, then this confusion goes away.